app/src/lib/shell.ts

```diff
@@ -36,144 +30,39 @@ export function shellNeedsPatching(process: NodeJS.Process): boolean {
   return __DARWIN__ && !process.env.PWD
 }
 
-type ShellResult = {
-  stdout: string
-  error: Error | null
-}
-
 /**
- * Gets a dump of the user's configured shell environment.
- *
- * @returns the output of the `env` command or `null` if there was an error.
- */
-async function getRawShellEnv(): Promise<string | null> {
-  const shell = getUserShell()
-
-  const promise = new Promise<ShellResult>(resolve => {
-    let child: ChildProcess.ChildProcess | null = null
-    let error: Error | null = null
-    let stdout = ''
-    let done = false
-
-    // ensure we clean up eventually, in case things go bad
-    const cleanup = () => {
-      if (!done && child) {
-        child.kill()
-        done = true
-      }
-    }
-    process.once('exit', cleanup)
-    setTimeout(() => {
-      cleanup()
-    }, 5000)
-
-    child = ChildProcess.spawn(shell, ['-ilc', 'command env'], {
-      detached: true,
-      stdio: ['ignore', 'pipe', process.stderr],
-    })
-
-    const buffers: Array<Buffer> = []
-
-    child.on('error', (e: Error) => {
-      done = true
-      error = e
-    })
-
-    // If Node.js encounters a synchronous runtime error while spawning
-    // `stdout` will be undefined and the error will be emitted asynchronously
-    if (child.stdout) {
-      child.stdout.on('data', (data: Buffer) => {
-        buffers.push(data)
-      })
-    }
-
-    child.on('close', (code: number, signal) => {
-      done = true
-      process.removeListener('exit', cleanup)
-      if (buffers.length) {
-        stdout = Buffer.concat(buffers).toString('utf8')
-      }
-
-      resolve({ stdout, error })
-    })
-  })
-
-  const { stdout, error } = await promise
-
-  if (error) {
-    // just swallow the error and move on with everything
-    return null
-  }
-
-  return stdout
-}
-
-function getUserShell() {
-  if (process.env.SHELL) {
-    return process.env.SHELL
-  }
-
-  return '/bin/bash'
-}
-
-/**
- * Get the environment variables from the user's current shell and update the
- * current environment.
- *
- * @param updateEnvironment a callback to fire if a valid environment is found
+ * Update the current process's environment variables using environment
+ * variables from the user's shell, if they can be retrieved successfully.
  */
-async function getEnvironmentFromShell(
-  updateEnvironment: (env: IndexLookup) => void
-): Promise<void> {
-  if (__WIN32__) {
+export async function updateEnvironmentForProcess(): Promise<void> {
+  if (!__DARWIN__) {
```

Comment:
Shouldn't we keep the previous check? It seems [Atom also runs this code for Linux](https://github.com/atom/atom/blob/d016186110ffbe588a45d594803f9f9078804085/src/update-process-env.js#L11). While we don't actively support Linux, it'd be a pity if we "broke" it or make it more difficult for the fork to keep up 🤔 

What do you think?
```suggestion
  if (__WIN32__) {
```

Reply:
I don't think we would break it as we only run this method if `shellNeedsPatching` returns true and that method has always required darwin:

https://github.com/desktop/desktop/blob/24dc211c6fdda0a2d5844a806ffaf3ca7db52009/app/src/lib/shell.ts#L26-L37

Reply:
Ohhh! Right, all good then 😄